core/src/main/java/org/elasticsearch/plugins/ListPluginsCommand.java

```diff
@@ -51,6 +51,18 @@ protected void execute(Terminal terminal, OptionSet options) throws Exception {
         try (DirectoryStream<Path> stream = Files.newDirectoryStream(env.pluginsFile())) {
             for (Path plugin : stream) {
                 terminal.println(plugin.getFileName().toString());
+                PluginInfo info = null;
+                try { 
+                    info = PluginInfo.readFromProperties(env.pluginsFile().resolve(plugin.toAbsolutePath()));
+                    terminal.println(Terminal.Verbosity.VERBOSE, info.toString());
+                }
+                catch(IOException ex) {
+                    terminal.println(Terminal.Verbosity.VERBOSE, "Plugin properties file missing for plugin");
+                }
+                catch (Exception ex){
```

Comment:
I dont think we should add leniency here. This will only hide the actual problem. Let the exception propagate.